### ISSUE TITLE
Feature request: drop dependency for Perl in `key-bindings.bash`

### ISSUE THREAD

**Issue by jlebon** [junegunn/fzf#3077]
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
- [x] I have searched through the existing issues

## Info

- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [x] bash
    - [ ] zsh
    - [ ] fish

## Problem / Steps to reproduce

`__fzf_history__()` in `key-bindings.bash` requires Perl. In more minimal environments, Perl may not be installed. It would be nice to be able to drop that dependency. Looking at [the exact code there](https://github.com/junegunn/fzf/blob/6b207bbf2bd916d06586cdbbf980ae507c01f2b2/shell/key-bindings.bash#L54), it seems like it's just doing some frequency counting of the bash command history. Could we have that be built into fzf perhaps and gated behind a flag?

**Comment by glensc** [comment]
this doesn't seem like big deal to rewrite in awk or so

**Comment by jlebon** [comment]
Looks like the latest effort is stuck on performance issues: https://github.com/junegunn/fzf/pull/3313#issuecomment-1573049093. I added a suggestion there.

**Comment by step-** [comment]
Author of #3395 (closed), #3310 (bash) and #3313 (awk) here. I read your suggestion in #3310. You're suggesting to implement an ad-hoc filter that would do what the perl filter (or my awk filter) does now. Am I right?

**Comment by jlebon** [comment]
> Author of #3395 (closed), #3310 (bash) and #3313 (awk) here. I read your suggestion in #3310. You're suggesting to implement an ad-hoc filter that would do what the perl filter (or my awk filter) does now. Am I right?

Right, I'm suggesting implementing the filter in golang behind some hidden switch. But I'm not a maintainer :) I would try to get @junegunn's opinion on it before doing this.

**Comment by step-** [comment]
can be closed due to #3313 merge

**Comment by step-** [comment]
TLDR; <https://github.com/junegunn/fzf/pull/3313#issuecomment-1573049093>

**Comment by glensc** [comment]
@kaddkaka broken about locale, add `LC_ALL=C` before perl command to mute them. probably worth sending PR to this repo too.

**Comment by kaddkaka** [comment]
> @kaddkaka broken about locale, add `LC_ALL=C` before perl command to mute them. probably worth sending PR to this repo too.

Thanks.I'm not confident to do a PR because I don't understand why that would work.

**Comment by glensc** [comment]
you can test that with your environment it no longer prints locale init errors:

```sh
perl -e 1
```

```sh
LC_ALL=C perl -e 1
```

perl would need locales as some code depends on locale data. regexes? but the current script doesn't seem to contain something that would need locales, so it seems safe :)
- https://github.com/step-/fzf/blob/341c13e99c8f3f371643c80ddb7c1b8217ded407/shell/key-bindings.bash#L55
